stl/inc/stacktrace

```diff
@@ -0,0 +1,347 @@
+// stacktrace standard header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _STACKSTRACE_
+#define _STACKSTRACE_
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+
+#if !_HAS_CXX23
+#pragma message("The contents of <stacktrace> are available only with C++23 or later.")
+#else // ^^^ !_HAS_CXX23 / _HAS_CXX23 vvv
+
+#include <cstdint>
+#include <string>
+#include <type_traits>
+#include <vector>
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+using _Stacktrace_string_fill_callback = size_t (*)(char*, size_t, void* _Context);
+
+using _Stacktrace_string_fill = size_t (*)(size_t, void* _String, void* _Context, _Stacktrace_string_fill_callback);
+
+inline size_t _Stacktrace_string_fill_impl(
+    const size_t _Size, void* const _String, void* const _Context, const _Stacktrace_string_fill_callback _Callback) {
+    if (_Callback) {
+        struct _Stacktrace_string_fill_op {
+            size_t operator()(char* _Data, size_t _Size) {
+                return _Callback(_Data, _Size, _Context);
+            }
+
+            _Stacktrace_string_fill_callback _Callback;
+            void* _Context;
+        };
+
+        static_cast<_STD string*>(_String)->resize_and_overwrite(
+            _Size, _Stacktrace_string_fill_op{_Callback, _Context});
+        return static_cast<_STD string*>(_String)->size();
+    } else {
+        return static_cast<_STD string*>(_String)->capacity();
+    }
+}
+
+_EXTERN_C
+unsigned short __stdcall __std_stacktrace_capture(unsigned long _FramesToSkip, unsigned long _FramesToCapture,
+    void** _BackTrace, unsigned long* _BackTraceHash) noexcept;
+
+// Some of these exports may throw (They would propagate bad_alloc potentially thrown from string::resize_and_overwrite)
+
+// clang-format off
+void __stdcall __std_stacktrace_address_to_string(
+    const void* _Address, void* _Str, _Stacktrace_string_fill) noexcept(false);
+
+void __stdcall __std_stacktrace_description(
+    const void* _Address, void* _Str, _Stacktrace_string_fill) noexcept(false);
+
+void __stdcall __std_stacktrace_source_file(
+    const void* _Address, void* _Str, _Stacktrace_string_fill) noexcept(false);
+
+[[nodiscard]] unsigned __stdcall __std_stacktrace_source_line(const void* _Address) noexcept;
+
+void __stdcall __std_stacktrace_to_string(const void* _Addresses, size_t _Size, void* _Str,
+    _Stacktrace_string_fill) noexcept(false);
+
+[[nodiscard]] void* __stdcall __std_stacktrace_get_debug_interface() noexcept;

```

Comment:
These should be `_NODISCARD` as we're still using that macro (although we might unconditionally use the attribute in the future).

Additionally, we conventionally always say `unsigned int` instead of `unsigned`.